src/terminal/index.ts

```diff
@@ -326,21 +316,66 @@ class TerminalWidget extends Widget {
     let rows = Math.floor(height / this._rowHeight) - 1;
     let cols = Math.floor(width / this._colWidth) - 1;
     this._term.resize(cols, rows);
-    this._session.send({
-      type: 'set_size',
-      content: [rows, cols, height, width]
-    });
-    this._dirty = false;
-    this.update();
+    this._sessionSize = [rows, cols, height, width];
+    this._setSessionSize();
+    this._needsResize = false;
+  }
+
+  /**
+   * Send the size to the session.
+   */
+  private _setSessionSize(): void {
+    if (this._session) {
+      this._session.send({
+        type: 'set_size',
+        content: this._sessionSize
+      });
+    }
+  }
+
+  /**
+   * Set the stylesheet.
+   */
+  private _setStyle(): void {
+    // Set the fg and bg colors of the terminal and cursor.
+    this._sheet.innerHTML = (`
+      #${this.node.id} {
+        background: ${this._background};
+        color: ${this._color};
+      }
+      #${this.node.id} .xterm-viewport, #${this.node.id} .xterm-rows {
+        background-color: ${this._background};
+        color: ${this._color};
+      }
+      #${this.node.id} .terminal.focus .terminal-cursor.blinking {
+          animation: ${this.node.id}-blink-cursor 1.2s infinite step-end;
+      }
+      @keyframes ${this.node.id}-blink-cursor {
+          0% {
+              background-color: ${this._color};
+              color: ${this._background};
+          }
+          50% {
+              background-color: transparent;
+              color: ${this._color};
+          }
+      }
+    `);
+    this._needsStyle = false;
   }
 
   private _term: Xterm = null;
   private _sheet: HTMLElement = null;
   private _dummyTerm: HTMLElement = null;
   private _fontSize = -1;
-  private _dirty = false;
+  private _needsSnap = true;
+  private _needsResize = true;
+  private _needsStyle = true;
   private _rowHeight = -1;
   private _colWidth = -1;
+  private _offsetWidth = -1;
+  private _offsetHeight = -1;
+  private _sessionSize: number[] = [1, 1, 1, 1];
```

Comment:
This is *always* an n-tuple with four element. Do we have a convention for these, like declaring it:
```
private _sessionSize: [number, number, number, number] = [1, 1, 1, 1];
```
It's fine the way it is, obviously. Wondering what your thoughts are.